commit 477c1359b217cdc052a7c9f83cae9c894396894c
Author: Jan Kratochvil <jan.kratochvil@redhat.com>
Date:   Mon Jun 28 22:03:31 2010 +0000

    gdb/
    	* dwarf2read.c (read_structure_type) <fi.typedef_field_list>: Call
    	ALLOCATE_CPLUS_STRUCT_TYPE.

### a/gdb/ChangeLog
### b/gdb/ChangeLog
## -1,3 +1,8 @@
+2010-06-28  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
+	* dwarf2read.c (read_structure_type) <fi.typedef_field_list>: Call
+	ALLOCATE_CPLUS_STRUCT_TYPE.
+
 2010-06-28  Phil Muldoon  <pmuldoon@redhat.com>
             Tom Tromey  <tromey@redhat.com>
             Thiago Jung Bauermann  <bauerman@br.ibm.com>
## -27,8 +32,6 @@
 	* python/py-inferior.c: New File.
 	* python/py-infthread.c: New File.
 
-
-
 2010-06-28  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
 	* c-typeprint.c (c_type_print_base): For no fields check include also
--- a/gdb/dwarf2read.c
+++ b/gdb/dwarf2read.c
@@ -5327,6 +5327,7 @@ read_structure_type (struct die_info *die, struct dwarf2_cu *cu)
 	{
 	  int i = fi.typedef_field_list_count;
 
+	  ALLOCATE_CPLUS_STRUCT_TYPE (type);
 	  TYPE_TYPEDEF_FIELD_ARRAY (type)
 	    = TYPE_ALLOC (type, sizeof (TYPE_TYPEDEF_FIELD (type, 0)) * i);
 	  TYPE_TYPEDEF_FIELD_COUNT (type) = i;
